### ISSUE TITLE
ion-toast should expose content margin/padding properties

### ISSUE THREAD

**Issue by distante** [ionic-team/ionic#17669]
# Feature Request

**Ionic version:**
[x] **4.0.2**

**Describe the Feature Request**
Currently the `ion-toast` does not expose any [custom CSS property](https://ionicframework.com/docs/api/toast#css-custom-properties) to set some kind of margin/padding to what is rendered inside their Shadow Dom.  There are some times when this toast needs to be at the bottom but when we use banner ads on our apps it gets - obviously- rendered below the banner ad.

**Describe Preferred Solution**
Expose some `--margin`, or `--padding` so we can move the Toast container 

**Describe Alternatives**
What I am doing right now is move the complete `ion-toast` using `translateY`

**Comment by brandyscarney** [comment]
Thanks for the issue! So this is a bit tricky because the toast is positioned by the animation. If we were to expose and change the margin it would mess up the default animation. I took a gif to show what I mean, this is applying `160px` margin bottom to the iOS toast (I added a ton of margin for emphasis):

![toast with margin](http://g.recordit.co/EFRxLeHV4n.gif)

Notice how when it leaves it disappears before reaching the bottom? Here it is normally:

![default toast](http://g.recordit.co/VEDpVudgJY.gif)

I'll add this as a feature request to make it easier to accomplish, but right now it will require passing a custom animation to the toast to position it. We have some examples on customizing the animation in the [utilities -> config section](https://ionicframework.com/docs/utilities/config#customizing-animations).
